recaf-core/src/main/java/me/coley/recaf/evaluation/ExpressionEvaluator.java

```diff
@@ -0,0 +1,40 @@
+package me.coley.recaf.evaluation;
+
+import com.github.javaparser.ast.expr.*;
+
+import java.util.Optional;
+
+public class ExpressionEvaluator {
+	private ExpressionEvaluator() {}
+
+	public static Optional<Number> evaluate(BinaryExpr binOp) {
+		Optional<Number> leftValue = evaluate(binOp.getLeft());
+		Optional<Number> rightValue = evaluate(binOp.getRight());
+		if (leftValue.isPresent() && rightValue.isPresent()) {
+			return NumberEvaluator.evaluate(leftValue.get(), binOp.getOperator(), rightValue.get());
+		}
+		return Optional.empty();
+	}
+
+	public static Optional<Number> evaluate(UnaryExpr unaryExpr) {
+		return evaluate(unaryExpr.getExpression())
+			.flatMap(n -> NumberEvaluator.evaluate(unaryExpr.getOperator(), n));
+	}
+
+	public static Optional<Number> evaluate(LiteralExpr literalExpr) {
+		return literalExpr.toIntegerLiteralExpr().map(IntegerLiteralExpr::asNumber)
+			// No byte, short, float
+			.or(() -> literalExpr.toDoubleLiteralExpr().map(DoubleLiteralExpr::asDouble))
+			.or(() -> literalExpr.toLongLiteralExpr().map(LongLiteralExpr::asNumber));
+	}
+
+	public static Optional<Number> evaluate(Expression expr) {
+		//un-enclosing parenthesis
+		if (expr instanceof EnclosedExpr) return evaluate(((EnclosedExpr) expr).getInner());
```

Comment:
Optimization nit-pick: Can do something like `while(expr is Enclosed) expr = extract(expr)` so that we can handle all tiers of enclosure in a single method frame.

Reply:
Thought of it, but expr needs to be effectively final for the lambda later.

Reply:
You can copy it to a second final variable after.

Reply:
Ok, will do it then.